lib/ember_cli/shell.rb

```diff
@@ -57,13 +57,14 @@ def test
     delegate :run, :run!, to: :runner
 
     def invalid_ember_dependencies?
-      !run("#{paths.ember} version")
+      !run("#{paths.ember} version").success?
     rescue DependencyError
       false
     end
 
     def clean_ember_dependencies!
-      ember_dependency_directories.select(&:exist?).each(&:rmtree)
+      dirs = ember_dependency_directories.select(&:exist?).flat_map(&:children)
+      dirs.each(&:rmtree)
```

Comment:
Thanks for opening this PR @krasnoukhov !

I apologize for letting this method get a little out of hand.

What do you think about extracting some private helper methods to more clearly communicate this method's intent?

Would this be a good start?:

```ruby
def ember_dependency_directories
  [
    paths.node_modules,
    paths.bower_components,
  ].select(&:exist?)
end

def clean_ember_dependencies!
  ember_dependency_directories.flat_map(&:children).each(&:rmtree)
end
```